server/controllers/auth.js

```diff
@@ -0,0 +1,128 @@
+/* eslint-disable sort-keys */
+/* eslint-disable consistent-return */
+const passport = require('passport');
+const dotenv = require('dotenv');
+const util = require('util');
+const { URL } = require('url');
+const querystring = require('querystring');
+const HttpStatus = require('http-status-codes');
+
+const StatusText = require('./constants');
+
+const { SUCCESS, ERROR, FAIL } = StatusText;
+const { OK, INTERNAL_SERVER_ERROR, UNAUTHORIZED } = HttpStatus;
+// load user model
+// const User = require('../models/User');
+
+dotenv.config();
+
+module.exports = {
+    callback: async (req, res, next) => {
```

Comment:
no need for this to be `async/await`. You are using a callback in `passport.authenticate`. For you to use the `async/await` pattern you have to convert `passport.authenticate` to an async function using `util.promisify` (http://2ality.com/2017/05/util-promisify.html)

Reply:
I can't wrap my head around Util.Promisify. So i just removed the async and await